ambry-cloud/src/main/java/com.github.ambry.cloud/azure/AzureCloudDestination.java

```diff
@@ -309,15 +310,14 @@ public boolean updateBlobExpiration(BlobId blobId, long expirationTime) throws C
       metadataList = getBlobMetadataChunked(blobIds);
     }
 
-    return metadataList.stream().collect(Collectors.toMap(m -> m.getId(), Function.identity()));
+    return metadataList.stream().collect(Collectors.toMap(CloudBlobMetadata::getId, Function.identity()));
   }
 
   private List<CloudBlobMetadata> getBlobMetadataChunked(List<BlobId> blobIds) throws CloudStorageException {
     if (blobIds.isEmpty() || blobIds.size() > ID_QUERY_BATCH_SIZE) {
       throw new IllegalArgumentException("Invalid input list size: " + blobIds.size());
     }
-    String quotedBlobIds =
-        String.join(",", blobIds.stream().map(s -> '"' + s.getID() + '"').collect(Collectors.toList()));
+    String quotedBlobIds = blobIds.stream().map(s -> '"' + s.getID() + '"').collect(Collectors.joining(","));
     String query = String.format(BATCH_ID_QUERY_TEMPLATE, quotedBlobIds);
     String partitionPath = blobIds.get(0).getPartition().toPathString();
     try {
```

Comment:
This may be an action item for the future, but we should consider if we can get rid of a lot of the string concat/formatting when running these queries.

There is an alternate constructor for `SqlQuerySpec` that is used in `getDeadBlobs`: `SqlQuerySpec(String queryText, SqlParameterCollection parameters)`

Reply:
Nevermind, I found that the client doesn't support making list `SqlParameter` that could be used for in clauses: https://github.com/Azure/azure-cosmosdb-java/issues/60